dev/domain-controller/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -0,0 +1,210 @@
+    then
+        echo "Driver is installed successfully"
+    else
+        echo "failed to install Nvidia driver. Will create a script to install driver when machine boots up"
+        file_path=/root/install_driver.sh
+        cat <<EOF >$file_path
+#!/bin/bash
+if [ -e /root/.first_boot ]
+then
+    if [ -e /root/.second_boot ]
+    then
+        exit 0
+    else
+        touch /root/.second_boot
+        sudo $FILE_LOCATION -Z -X -s        
+    fi
+else
+    touch /root/.first_boot
+    sudo $FILE_LOCATION -Z -X -s
+    (sleep 2;  sudo shutdown -f -r +0)&
+fi
+EOF
+        sudo chmod 744 $file_path
+        (crontab -l 2>/dev/null; echo "@reboot $file_path") | crontab -
+        # let's define exit code 104 for this case
+        #exit 104
+    fi
+}
+
+# the first argument is the Registration Code of PCoIP agent
+REGISTRATION_CODE=$1
+# the second argument is the agent type
+AGENT_TYPE=$2
+# the third argument is the VM name
+VM_NAME=$3
+# the forth argument is the domain name
+DOMAIN_NAME=$4
+# the fifith argument is the username
+USERNAME=$5
+# the sixth argument is the password
+PASSWORD=$6
+# the seventh argument is the domain group to join
+GROUP=$7
+
+# Make sure Linux OS is up to date
+echo "--> Updating Linux OS to latest"
+# Exclude WALinuxAgent due to it failing to update from within an Azure Custom Script
+sudo yum -y update --exclude=WALinuxAgent
+
+# If it's graphic agent, install Nvidia Driver
+case "$AGENT_TYPE" in 
+    "Graphics")
+        install_driver
+        AGENT_TYPE='graphics'
+        ;;
+    "Standard")
+        AGENT_TYPE='standard'
+        ;;   
+    *)
+        echo "unknown agent type $AGENT_TYPE."
+        # let's define exit code 105 for this case
+        exit 105
+        ;;       
+esac
+
+# Join domain
+echo "-->Install required packages to join domain"
+sudo yum -y install sssd realmd oddjob oddjob-mkhomedir adcli samba-common samba-common-tools krb5-workstation openldap-clients policycoreutils-python
+sudo systemctl enable sssd
+
+echo "-->Joining the domain"
+echo $PASSWORD | sudo realm join --user=$USERNAME $DOMAIN_NAME
+
+echo "-->Configuring settings"
+sudo sed -i '$ a\dyndns_update = True\ndyndns_ttl = 3600\ndyndns_refresh_interval = 43200\ndyndns_update_ptr = True\nldap_user_principal = nosuchattribute' /etc/sssd/sssd.conf
+sudo sed -c -i "s/\(use_fully_qualified_names *= *\).*/\1False/" /etc/sssd/sssd.conf
+sudo sed -c -i "s/\(fallback_homedir *= *\).*/\1\/home\/%u/" /etc/sssd/sssd.conf
+sudo domainname $VM_NAME.$DOMAIN_NAME
+echo "%$DOMAIN_NAME\\\\Domain\\ Admins ALL=(ALL) ALL" > /etc/sudoers.d/sudoers
+
+echo "-->Registering with DNS"
+DOMAIN_UPPER=$(echo "$DOMAIN_NAME" | tr "[a-z]" "[A-Z]")
+IP_ADDRESS=$(hostname -I | grep -Eo '10.([0-9]*\.){2}[0-9]*')
+echo $PASSWORD | sudo kinit $USERNAME@$DOMAIN_UPPER
+touch dns_record
+echo "update add $VM_NAME.$DOMAIN_NAME 600 a $IP_ADDRESS" > dns_record
+echo "send" >> dns_record
+sudo nsupdate -g dns_record
+
+# Install the EPEL repository
```

Comment:
Does EPEL stand for anything interesting? Why do we need it? (Maybe can flesh out the comment a bit more.)

Reply:
This script was taken from 1 click deployment. I'm not sure why it was installing EPEL repository. @fwang-teradici , Do you know why we need this?

Reply:
From the document to install PCoIP Agent on REPL
https://techsupport.teradici.com/ics/support/kbanswer.asp?deptID=15164&task=knowledge&questionID=3147

Reply:
@bill-dall: EPEL: Extra Packages for Enterprise Linux

Reply:
Okay thank.